tests/manage/z_cluster/cluster_exp_helpers.py

```diff
@@ -0,0 +1,215 @@
+
+    """
+
+    dir_name = "cluster_copy_ops_" + uuid4().hex
+    cmd = "" + "mkdir /mnt/" + dir_name + ""
+    copy_pod.exec_sh_cmd_on_pod(cmd, sh="bash")
+
+    # cp ceph.tar.gz to 1 dir
+
+    cmd1 = f"mkdir /mnt/{dir_name}/copy_dir1"
+    cmd2 = f"cp /mnt/ceph.tar.gz /mnt/{dir_name}/copy_dir1/"
+    copy_pod.exec_sh_cmd_on_pod(cmd1, sh="bash")
+    copy_pod.exec_sh_cmd_on_pod(cmd2, sh="bash")
+
+    # check md5sum
+    # since the file to be copied is from a wellknown location, we calculated its md5sum and found it to be:
+    # 016c37aa72f12e88127239467ff4962b. We will pass this value to the pod to see if it matches that of the same
+    # file copied in different directories of the pod.
+    # (we could calculate the md5sum by downloading first outside of pod and then comparing it with that of pod.
+    # But this will increase the execution time as we have to wait for download to complete once outside pod and
+    # once inside pod)
+    md5sum_val_expected = "016c37aa72f12e88127239467ff4962b"
+
+    # We are not using the pod.verify_data_integrity with the fedora dc pods as of now for the following reason:
+    # verify_data_integrity function in pod.py calls check_file_existence which in turn uses 'find' utility to see
+    # if the file given exists or not. In fedora pods which this function mainly deals with, the 'find' utility
+    # doesn't come by default. It has to be installed. While doing so, 'yum install findutils' hangs.
+    # the link "https://forums.fedoraforum.org/showthread.php?320926-failovemethod-option" mentions the solution:
+    # to run "sed -i '/^failovermethod=/d' /etc/yum.repos.d/*.repo". This command takes at least 6-7 minutes to
+    # complete. If this has to be repeated on 24 pods, then time taken to complete may vary between 20-30 minutes
+    # even if they are run in parallel threads.
+    # Instead of this if we use shell command: "md5sum" directly we can reduce the time drastically. And hence we
+    # are not using verify_data_integrity() here.
+
+    # for i in range(1, 2):
+    cmd = f"md5sum /mnt/{dir_name}/copy_dir1/ceph.tar.gz"
+    output = copy_pod.exec_sh_cmd_on_pod(cmd, sh="bash")
+    md5sum_val_got = output.split("  ")[0]
+    logger.info(f"#### md5sum obtained for pod: {copy_pod.name} is {md5sum_val_got}")
+    logger.info(f"#### Expected was: {md5sum_val_expected}")
+    if md5sum_val_got != md5sum_val_expected:
+        logging.info(f"***** md5sum check FAILED. expected: {md5sum_val_expected}, but got {md5sum_val_got}")
+        cmd = "" + "ls -lR /mnt" + ""
+        output = copy_pod.exec_sh_cmd_on_pod(cmd, sh="bash")
+        logging.info(f"ls -lR /mnt output = {output}")
+        return False
+
+    logging.info("#### Data Integrity check passed")
+
+    # Remove the directories - clean up
+    cmd = f"rm -rf /mnt/{dir_name}/copy_dir1"
+    logging.info(f"#### command to remove = {cmd}")
+    copy_pod.exec_sh_cmd_on_pod(cmd, sh="bash")
+
+    return True
+
+
+class ClusterFiller():
```

Comment:
this class seems to be relevant for other test modules, apart from cluster related ones. Can this be moved to ocs_ci/ dir?

Reply:
I'm thinking to take it in follow up pr